internal/provider/resource_stack_component.go

```diff
@@ -96,12 +97,28 @@ func resourceStackComponentCreate(ctx context.Context, d *schema.ResourceData, m
 	}
 
 	// Create the component request
+	configMap := make(map[string]interface{})
+	for k, v := range d.Get("configuration").(map[string]interface{}) {
+		stringValue := v.(string)
+		// Handle known JSON fields that need parsing
+		if k == "labels" {
```

Comment:
Unfortunately, this approach is not maintainable. We'll have to keep adding fields here whenever we discover that something doesn't work.

What we need is a generic approach, maybe try allowing more than strings as values in the configuration dictionary (i.e. also integers, booleans, lists and dictionaries).